tests/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImplTest.java

```diff
@@ -216,6 +228,10 @@ public void testOutputMode() {
 
     outputMode = ConsoleRunnerImpl.OutputMode.NONE;
     output = runTest(OutputModeTest.class, true);
+    assertThat(output, containsString("Output in classSetUp"));
+    assertThat(output, not(containsString("Output in setUp")));
+    assertThat(output, not(containsString("Output in tearDown")));
+    assertThat(output, not(containsString("Output in classTearDown")));
```

Comment:
it still feels weird to me that class setup is included in NONE mode. I would expect NONE to have no output.

Reply:
The methods annotated with `@BeforeClass` are called before we get the first `testStarted` callback on the `RunListener` so we haven't set up an alternate stream for the output to go to.  I assume output in an `@BeforeClass` method would be rare and I was mainly adding the test to show what the behavior is.  You could probably fix this by swapping the streams in the `testRunStarted` callback but I didn't want to add more changes to this PR for that edge case.

Reply:
Makes sense to me.